.github/workflows/docs.yml

```diff
@@ -0,0 +1,194 @@
+      - uses: dorny/paths-filter@v2
+        if: github.event_name == 'pull_request' 
+        id: filter
+        with:
+          filters: | # TODO: Remove '|(.github/workflows/docs.yml)' once workflow gets tested
+            path-filter:
+              - './(((docs|solidity)/**)|(.github/workflows/docs.yml))'
+
+  docs-tex:
+    runs-on: ubuntu-latest
+    container: keepnetwork/texlive:3
+    needs: docs-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    defaults:
+      run:
+        working-directory: ./docs
+    env:
+      DOCS_DIR: ${{ github.workspace }}/tmp/docs
+    steps:
+      - uses: actions/checkout@v2
+
+      # Step generates files based on Makefile config. Current config produces
+      # a number of .png files, which are then used to generate 
+      # relay-states.pdf. We want to publish only the resulting .pdf file.
+      - name: Generate TeX document files
+        run: |
+          make clean
+          make docs
+          mkdir -p $DOCS_DIR
+          mv relay-states.pdf $DOCS_DIR
+
+      # A push event is a master merge; deploy to primary bucket.
+      - if: github.event_name == 'push'
+        name: Upload generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core-wip2 # TODO: remove `-wip2` after tests
+          build-folder: tmp/docs
+
+      # A pull_request event is a PR; deploy to preview bucket.
+      - if: github.event_name == 'pull_request'
+        name: Upload preview of generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core/${{ github.head_ref }}-wip2 # TODO: remove `-wip2` after tests
+          build-folder: tmp/docs
+    
+  docs-solidity:
+    runs-on: ubuntu-latest
+    needs: docs-detect-changes
+    continue-on-error: true
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Generate solidity docs
+        run: |
+          cd solidity && npm install
+          node_modules/.bin/truffle compile
+          mkdir -p ${{ github.workspace }}/docs/solidity
+          node scripts/generate-api-docs.js > ${{ github.workspace }}/docs/solidity/index.adoc
+
+      - name: Upload a Build Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: Solidity Docs
+          path: ${{ github.workspace }}/docs/solidity/index.adoc
+
+  docs-html:
+    runs-on: ubuntu-latest
+    needs: [docs-detect-changes, docs-solidity]
```

Comment:
We want to run `docs-html` and `docs-pdf` jobs even if `docs-solidity` failed ([`if: always()`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-not-requiring-dependent-jobs-to-be-successful)).

Reply:
See 6f8839b5b10e9d059ba583a702e503c131e879b2. I also removed the `continue-on-error: true` from `docs-solidity`.

Reply:
Does this also make it run if `docs-detect-changes` reports no changes though?

Reply:
You're right, it does. I've decided to provide additional conditions - jobs will be executed even if earlier jobs fail, but only if `github.event_name != 'pull_request' || needs.docs-detect-changes.outputs.path-filter == 'true'` criterium is met.
I also considered setting `continue-on-error` for `docs-solidity` job and removing `if` conditions from `docs-html` and `docs-pdf` jobs, but that would not mark the workflow as failed in case of `docs-solidity` job failure (making it possible for us to not notice the failure in some situations).